spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -1757,7 +1868,11 @@ trait OptimisticTransactionImpl extends TransactionalWrite
           committed = true
           return (commitVersion, postCommitSnapshot, updatedCurrentTransactionInfo)
         } catch {
-          case _: FileAlreadyExistsException if preCommitCommitStoreOpt.isEmpty =>
+          case _: FileAlreadyExistsException if isFsToMcCommit =>
+            // Don't retry if this commit tries to upgrade the table from filesystem to managed
```

Comment:
Why not retry this?

Reply:
The retry would need us to register again with the commit-owner. Currently we register only once in the begiinning on txn start.